src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Utility.cs

```diff
@@ -1394,6 +1394,17 @@ private static bool VectorContainsNonAsciiChar(Vector128<byte> asciiVector)
             }
         }
 
+        /// <summary>
+        /// Stores to lower 64bits of <paramref name="byteVector"/> to memory destination of <paramref name="bytePtr"/>[<paramref name="elementOffset"/>]
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static unsafe void StoreLower(Vector128<byte> byteVector, ref byte bytePtr, nuint elementOffset)
+        {
+            // Below code translates to a single write on x86 (for both 32 and 64 bit)
+            // - we use double instead of long so that the JIT writes directly to memory without intermediate (register or stack in case of 32 bit)
+            Unsafe.WriteUnaligned<double>(ref Unsafe.Add(ref bytePtr, elementOffset), byteVector.AsDouble().ToScalar());
```

Comment:
LGTM, but I don't think that comment is needed because it is also expected to be a single instruction on arm. Also I'm not sure double is any better than long here, jit is expected to do the right thing anyway.

Reply:
I can believe that the double hack helps on 32-bit x86. We do not pay as much attention to the codegen quality on 32-bit x86 and there are definitely issues. I do not think it is worth it to be adding workarounds like this for x86 quality issues to CoreLib.  If issues like this one are important to fix, it would be better to fix it in the JIT.

Reply:
Ah, I missed the fact that the same code with `long` generates bad codegen on x86, I hoped JIT would emit the same as for double 🙁 @jkotas do you mean the whole helper call is not needed? (it is needed because the original code was using Vector64 that we don't recommend using on x86/64) or you're fine with changing this to `long` and remove notes?

Reply:
![image](https://user-images.githubusercontent.com/523221/236651736-eacb26da-8f6b-46bc-9ad6-8a2734c47d1f.png)

(32bit)

Reply:
> @jkotas do you mean the whole helper call is not needed?

I assume that the helper call is needed to avoid `Vector64` that does not produce good code on x64. Without the helper call., the alternative sequence that avoids `Vector64` would have to be manually inlined in every place.